src/datajunction/cli/add_database.py

```diff
@@ -0,0 +1,45 @@
+"""
+Add a database to an existing repository
+"""
+
+import logging
+from pathlib import Path
+from typing import Optional
+
+import yaml
+
+from datajunction.errors import DJError, DJException, ErrorCode
+from datajunction.models.database import Database
+
+_logger = logging.getLogger(__name__)
+
+def create_databases_dir(repository: Path):
+    """
+    Creates a databases directory in the repository if one doesn't already exist
+    """
+    database_dir_path = repository / Path("databases")
+    database_dir_path.mkdir(exist_ok=True)
+
+async def run(repository: Path, database: str, uri: str, description: Optional[str] = None, read_only: Optional[bool] = None, cost: Optional[float] = None) -> None:
+    """
+    Add a database.
+    """
+    create_databases_dir(repository)
+    db_config_path = repository / Path(f"databases/{database}.yaml")
```

Comment:
If you change `create_databases_dir` to return the path you can do:

```suggestion
    databases_dir = create_databases_dir(repository)
    db_config_path = (databases_dir / database).with_suffix(".yaml")
```